endOfEpoch/metachain/trigger.go

```diff
@@ -0,0 +1,105 @@
+package metachain
+
+import (
+	"github.com/ElrondNetwork/elrond-go/config"
+	"github.com/ElrondNetwork/elrond-go/core/logger"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/endOfEpoch"
+	"github.com/ElrondNetwork/elrond-go/ntp"
+)
+
+var log = logger.DefaultLogger()
+
+type ArgsNewMetaEndOfEpochTrigger struct {
+	Rounder     endOfEpoch.Rounder
+	SyncTimer   ntp.SyncTimer
+	GenesisTime time.Time
+	Settings    *config.EndOfEpochConfig
+	Epoch       uint32
+}
+
+type trigger struct {
+	epoch                         uint32
+	rounder                       endOfEpoch.Rounder
+	roundsPerEpoch                int64
+	roundsBetweenForcedEndOfEpoch int64
+	epochStartTime                time.Time
+	syncTimer                     ntp.SyncTimer
+}
+
+func NewEndOfEpochTrigger(args *ArgsNewMetaEndOfEpochTrigger) (*trigger, error) {
```

Comment:
comment for exported methods

Check all methods below

Reply:
done.